theano/sandbox/cuda/opt_util.py

```diff
@@ -42,7 +46,17 @@ def find_node(v, cls):
             return None
 
 
-def alpha_merge(cls, alpha_in, nd):
+def is_equal(var, val):
+    # Returns True if var is always equal to val (python value), False
+    # otherwise (including if var is not constant)
+    try:
+        v = get_scalar_constant_value(var)
+        return v == val
```

Comment:
Wouldn't that be covered by the NotScalarConstantError?  Also I do want this to work through dimshuffles and the like.